site/src/content/docs/best-practices/upgrading-zarf.mdx

```diff
@@ -0,0 +1,62 @@
+---
+title: "Upgrading Zarf"
+---
+
+Optimally the version of Zarf performing the `package create` and `package deploy` operations aligns to the version of [Init Package](/ref/init-package) that is installed within the cluster.
+
+Follow [CLI Installation](/getting-started/install) steps for installing updates to the version of binary installed on the machine executing zarf commands.
+
+# Upgrading the Init Package
+
+Upgrading the init package is a bit different from upgrading a regular application package because it establishes your cluster's foundational infrastructure. Follow these steps to upgrade your init package:
+
+## 1. Obtain the Updated Init Package
+
+- **Default Package:** If you're using the default package, run the following command to fetch the latest init package (if it's not already present locally):
+
+  ```bash
+  zarf tools download-init
+  ```
+
+  - **Validate:** This will download the associated init package to your current working directory unless `--output-directory` is specified.
+
+- **Manual:** The Init Package can also be manually downloaded from the Github releases to a storage location of your choice.
+
+  ```bash
+  VERSION="<target version>"; ARCH="<target architecture>"; curl -LO https://github.com/zarf-dev/zarf/releases/download/${VERSION}/zarf-init-${ARCH}-${VERSION}.tar.zst
+  ```
+
+## 2. Re-initialize the cluster
+
+- **Init:** The init process can be executed to upgrade a cluster from one version to another.
+
+  ```bash
+  zarf init --confirm
+  ```
+
+- **Deploy:** Init Packages can also be deployed directly with `package deploy` - allowing for explicit deployment specification and the use of [custom init packages](/tutorials/7-custom-init-packages).
+
+  ```bash
+  zarf package deploy <zarf-init-package tarball> --confirm
+  ```
+
+  **Note:** The use of package deploy for zarf init operations will not expose any configuration for external git/registry options.
+
+## 3. Validate Zarf Infrastructure
+
+- **Validate:** Ensure the following zarf resources are healthy:
+  - Registry (if deployed)
+  - Agent pods
+  - Git server (if deployed)
+
+  ```bash
+  zarf tools kubectl get pods -n zarf
+  ```
+
+# Special Considerations
+
+After the original `zarf init` has taken place - any following initialize actions will use the same `secret` for obtaining information provided by any `zarf init` flags.
```

Comment:
```suggestion
After the original `zarf init` has taken place - any following initialize actions will use the values in the `zarf-state` secret in the `zarf` namespace to obtain information for Zarf infrastructure such as the registry. For example, the `--registry-pull-password` flag is ignored on subsequent runs.
```
Thoughts on adding some more detail here? I think specifying that this is for flags related infrastructure and not flags like `--components` would make this more clear.